rundeckapp/grails-app/controllers/rundeck/controllers/ProjectController.groovy

```diff
@@ -2160,6 +2163,31 @@ key2=value'''
         }
         Properties currentProps = project.getProjectProperties() as Properties
 
+        //validate plugin property values
+        def projectScopedConfigs = frameworkService.discoverScopedConfiguration(configProps, "project.plugin")
+        projectScopedConfigs.each { String svcName, Map<String, Map<String, String>> providers ->
+            final pluginDescriptions = pluginService.listPluginDescriptions(svcName)
+            providers.each { String provider, Map<String, String> providerConfig ->
+                def desc = pluginDescriptions.find { it.name == provider }
+                if (desc) {
+                    def validation = pluginService.validatePluginConfig(svcName, provider, providerConfig, null)
+                    if (!validation.valid) {
+                        Validator.Report report = validation.report
+                        errors << (
+                                report.errors ?
+                                        "${provider} configuration was invalid: " + report.errors :
+                                        "${provider} configuration was invalid"
+                        )
+                        return apiService.renderErrorFormat(response,[
+                                status: HttpServletResponse.SC_BAD_REQUEST,
+                                message:errors,
+                                format:respFormat
+                        ])
```

Comment:
the response should not be rendered here, it is within ".each" loops and possibly would render multiple times, in addition the `return` statement will not exit the loops because they are not for loops but closures.  This should be deferred to outside of both loops.